### PR TITLE
Propose timeout adjuster refactoring [ECR-717]

### PR DESCRIPTION
**Pull request by stanislav-tkach**
See changelog for details. :upside_down_face:

### CODE REVIEW

File: exonum/src/blockchain/config.rs [hunk 1]
```diff
@@ -68,8 +71,12 @@ pub struct ConsensusConfig {
     pub txs_block_limit: u32,
     /// Maximum message length (in bytes).
     pub max_message_len: u32,
-    /// `TimeoutAdjuster` configuration.
-    pub timeout_adjuster: TimeoutAdjusterConfig,
+    /// Minimal propose timeout.
+    pub min_propose_timeout: Milliseconds,
+    /// Maximal propose timeout.
+    pub max_propose_timeout: Milliseconds,
+    /// Transactions threshold starting from which the `min_propose_timeout` value is used.
```
Comment (popzxc) [inline exonum/src/blockchain/config.rs]: Maybe "Amount of transactions in pool to start use `min_propose_timeout`"?

File: exonum/src/blockchain/config.rs [hunk 2]
```diff
@@ -79,17 +86,11 @@ impl ConsensusConfig {
     /// Checks if propose timeout is less than round timeout. Warns if fails.
     #[doc(hidden)]
     pub fn validate_configuration(&self) {
```
Comment (vldm) [inline exonum/src/blockchain/config.rs]: is "validate" apropriate prefix, for function that only warns about invalid config?

Comment (DarkEld3r) [reply exonum/src/blockchain/config.rs]: I'm not sure. :upside_down_face: What name do you propose? Is something like `check_configuration` better?

File: exonum/tests/testdata/config/config0_pub.toml [hunk 3]
```diff
@@ -4,6 +4,9 @@ peers_timeout = 10000
 round_timeout = 3000
 status_timeout = 5000
 txs_block_limit = 1000
+min_propose_timeout = 10
+max_propose_timeout = 200
+propose_timeout_threshold = 1
 
 [common.consensus_config.timeout_adjuster]
```
Comment (slowli) [inline exonum/tests/testdata/config/config0_pub.toml]: Are these sections in TOML configs still relevant?

File: exonum/src/blockchain/config.rs (hunk 2 above)
Comment (DarkEld3r) [reply exonum/src/blockchain/config.rs]: I will fix it in a separate pull request. :upside_down_face:

**Pull request merged**